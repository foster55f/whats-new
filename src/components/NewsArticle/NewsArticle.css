.newsArticle {
    box-sizing: border-box;
    border: 3px solid black;
    padding: 10px;
  }

  .img {
      width: 100%;
      height: 100%;
      object-fit: cover
  }