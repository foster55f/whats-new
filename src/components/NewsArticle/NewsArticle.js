import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({id, title, image, description, url}) => {
    return (
        <article className='newsArticle' id={id}>
            <div>
                <img className='img' src={image} alt={`of ${title}`}/>
            </div>
            <div>
                <h2>{title}</h2>
                <p>{description}</p>
                <a href={url}>Article Link</a>
            </div>
        </article>
    )
}

export default NewsArticle;