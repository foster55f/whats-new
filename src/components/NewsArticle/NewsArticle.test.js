import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('renders the title of the NewsArticle in <h2> tags', () => {
    const wrapper = shallow(<NewsArticle title="Bananas" />);
    const title = <h2>Bananas</h2>;

    expect(wrapper.contains(title)).toEqual(true);
  });
    
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
        title="Bananas"
        description="B A N A N A S"
        id={3}
        image="image"
        url="foster.com"
    />);

    expect(wrapper).toMatchSnapshot();
  });
});