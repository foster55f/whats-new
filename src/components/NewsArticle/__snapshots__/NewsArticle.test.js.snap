// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<article
  className="newsArticle"
  id={3}
>
  <div>
    <img
      alt="of Bananas"
      className="img"
      src="image"
    />
  </div>
  <div>
    <h2>
      Bananas
    </h2>
    <p>
      B A N A N A S
    </p>
    <a
      href="foster.com"
    >
      Article Link
    </a>
  </div>
</article>
`;
